File: aws-net-workshop/aws-net-workshop/examples/_00_CreateBucket.cs
using System;
using Amazon.S3;
using Amazon.S3.Model;
using System.Net;

namespace aws_net_workshop.examples
{
    public class _00_CreateBucket
    {
        public static void Main(string[] args)
        {
            // create the AWS S3 client
            AmazonS3Client s3 = AWSS3Factory.getS3Client();

            // create bucket request
            PutBucketRequest request = new PutBucketRequest()
            {
                BucketName = AWSS3Factory.S3_BUCKET
            };

            // create bucket - used for subsequent demos
            s3.PutBucket(request);

            // create bucket lising request
            ListObjectsRequest objects = new ListObjectsRequest()
            {
                BucketName = AWSS3Factory.S3_BUCKET
            };

            // get bucket lising to retrieve bucket name
            ListObjectsResponse result = s3.ListObjects(objects);

            // print bucket name for validation
            Console.WriteLine(string.Format("Successfully created bucket {0}.", result.Name));
            Console.ReadLine();
        }
    }
}

File: aws-java-workshop/src/main/java/com/emc/vipr/s3/sample/_00_CreateBucket.java
/*
 * Copyright 2013 EMC Corporation. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.txt
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */
package com.emc.vipr.s3.sample;

import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.model.ObjectListing;

public class _00_CreateBucket {

	public static void main(String[] args) throws Exception {
    	// create the AWS S3 Client
		AmazonS3 s3 = AWSS3Factory.getS3Client();

    	// create the bucket - used for subsequent demo operations
        s3.createBucket(AWSS3Factory.S3_BUCKET);
        
        // get bucket listing to retrieve the bucket name
        ObjectListing objects = s3.listObjects(AWSS3Factory.S3_BUCKET);

        // print bucket name for validation
    	System.out.println( String.format("Successfully created bucket [%s]", 
    			objects.getBucketName()));
    }
}
